### PR TITLE
docs: create key concept page for presets

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Create key concept page for presets: general overview, why you should use presets
- Existing "Shareable Config presets": instructions, preset hosting

## Context

Helps with #11237, but doesn't close it.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/config-presets.md [hunk 1]
```diff
@@ -25,19 +19,17 @@ Shareable config presets can only be used with the JSON format, other formats ar
...
 
-Renovate's configuration is self-documenting, because you can fill in the `"description"` field in all preset configs.
+## Extending from a preset
 
-## Implementation Approach
+To use a preset put it in an `extends` array within your Renovate config.
+Presets can be nested.
 
-In order to achieve these goals, preset configs allow for a very modular approach - preset configs can be as small as a partial package rule or as large as an entire configuration, like an ESLint config.
+QUESTION: Insert `renovate.json` example config with nested preset?
```
Comment (HonkingGoose) [inline docs/usage/config-presets.md:32]: We should add an example of a nested config preset. This way people can see what's valid nesting, and how it works.

Can you maintainers suggest a good example?

Comment (rarkins) [reply docs/usage/config-presets.md:32]: What do you mean by nested preset and what's the motivation?

Comment (viceice) [reply docs/usage/config-presets.md:32]: I think since we support files and paths for presets, we should deprecate nested presets (docs only note)

Comment (rarkins) [reply docs/usage/config-presets.md:32]: Preferring one preset per file?

Comment (viceice) [reply docs/usage/config-presets.md:32]: yes, would make things easier to explain in docs. 🤗

Comment (rarkins) [reply docs/usage/config-presets.md:32]: Agreed. We use nested ones in our internal though..

Comment (viceice) [reply docs/usage/config-presets.md:32]: > Agreed. We use nested ones in our internal though..

👀 we do? probably the (partial) auto generated presets?

Comment (HonkingGoose) [reply docs/usage/config-presets.md:32]: > What do you mean by nested preset and what's the motivation?

The _current_ docs mention "nested presets", check https://docs.renovatebot.com/config-presets/:

> Renovate's "config presets" are a convenient way to distribute config for reuse across multiple repositories. It is similar in design to ESLint's shareable configs, and can be used for whole repository configs and for individual rules. They are defined using the `extends` array within config and may also be nested.

That's why I thought we need an example on how to create a nested preset. 😉 

But if we can simplify by dropping support for "nesting" completely, that's even better. 🥳

Comment (HonkingGoose) [reply docs/usage/config-presets.md:32]: Do you want me to create an issue to "deprecate nested presets in the docs" then?

Comment (rarkins) [reply docs/usage/config-presets.md:32]: Yes please

**Review by rarkins — APPROVED** [review]

**Pull request merged**